wordpress-mobile/WordPress-Android
# The site settings title should update when the site title is updated

Fixes https://github.com/wordpress-mobile/WordPress-Android/issues/17775

The PR updates the screen title page when the user changes the site name

To test:

1. Launch the app.
2. Navigate to "My Site → Menu → Site Settings".
3. Tap "Site Title" and update the site title.
4. Notice title on the "Site Title" and the screen title at the top of the screen are both updated

## Regression Notes
1. Potential unintended areas of impact


2. What I did to test those areas of impact (or what existing automated tests I relied on)
NONE

3. What automated tests I added (or what prevented me from doing so)
Manual testing
PR submission checklist:


### Screenshots
**ISSUE**
https://user-images.githubusercontent.com/2471769/212954587-1065ee05-ef73-4e5f-ae8b-8f144826bc00.webm
**FIX**
https://user-images.githubusercontent.com/2725300/216615609-4deaaaf9-d05c-4bad-b8c4-bd3a441d465b.mp4
